subscriber_notifee.go

```diff
@@ -14,82 +16,197 @@ import (
 // subscriberNotifee implements network.Notifee and also manages the subscriber to the event bus. We consume peer
 // identification events to trigger inclusion in the routing table, and we consume Disconnected events to eject peers
 // from it.
-type subscriberNotifee IpfsDHT
-
-func (nn *subscriberNotifee) DHT() *IpfsDHT {
-	return (*IpfsDHT)(nn)
+type subscriberNotifee struct {
+	dht                                    *IpfsDHT
+	identifySub, updateSub, routabilitySub event.Subscription
 }
 
-func (nn *subscriberNotifee) subscribe(proc goprocess.Process) {
-	dht := nn.DHT()
-
-	dht.host.Network().Notify(nn)
-	defer dht.host.Network().StopNotify(nn)
+func newSubscriberNotifiee(dht *IpfsDHT) (*subscriberNotifee, error) {
+	bufSize := eventbus.BufSize(256)
```

Comment:
I'm reusing this for all of the different subscriptions. The PRs I took these from only gave 256 buffer size to `EvtPeerIdentificationCompleted` and the routability events, but no custom buffer size (so the default of 16) for `EvtPeerProtocolsUpdated`. I didn't see any logic as to how the numbers were chosen so I just gave them all 256 buffer sizes.

Reply:
@Stebalien can we still leave this as 256 shared between all three event types? It's probably more than enough right?